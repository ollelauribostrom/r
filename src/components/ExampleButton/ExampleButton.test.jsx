import React from 'react';
import { shallow } from 'enzyme';
import { ExampleButton } from '.';

describe('Tests for ExampleButton component', () => {
  it('renders the correct text', () => {
    const wrapper = shallow(<ExampleButton text="test" />);
    expect(wrapper.text()).toEqual('test');
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<ExampleButton text="test" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('invokes onClick handler when clicked', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<ExampleButton text="test" onClick={onClick} />);
    wrapper.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
