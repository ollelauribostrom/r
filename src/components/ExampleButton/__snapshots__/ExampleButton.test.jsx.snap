// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ExampleButton component matches snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  test
</button>
`;
